android/src/main/java/com/opentokreactnative/OTSessionManager.java

```diff
@@ -97,23 +97,38 @@ public void initPublisher(String publisherId, ReadableMap properties) {
         String resolution = properties.getString("resolution");
         Boolean publishAudio = properties.getBoolean("publishAudio");
         Boolean publishVideo = properties.getBoolean("publishVideo");
-
-        Publisher mPublisher = new Publisher.Builder(this.getReactApplicationContext())
-                                    .audioTrack(audioTrack)
-                                    .videoTrack(videoTrack)
-                                    .name(name)
-                                    .audioBitrate(audioBitrate)
-                                    .resolution(Publisher.CameraCaptureResolution.valueOf(resolution))
-                                    .frameRate(Publisher.CameraCaptureFrameRate.valueOf(frameRate))
-                                    .build();
+        String videoSource = properties.getString("videoSource");
+        if (videoSource.equals("screen")) {
+            View view = getCurrentActivity().getWindow().getDecorView().getRootView();
+            OTScreenCapturer capturer = new OTScreenCapturer(view);
+            Publisher mPublisher = new Publisher.Builder(this.getReactApplicationContext())
+                                        .audioTrack(audioTrack)
+                                        .videoTrack(videoTrack)
+                                        .name(name)
+                                        .audioBitrate(audioBitrate)
+                                        .resolution(Publisher.CameraCaptureResolution.valueOf(resolution))
+                                        .frameRate(Publisher.CameraCaptureFrameRate.valueOf(frameRate))
+                                        .capturer(capturer)                                        
+                                        .build();
+            mPublisher.setPublisherVideoType(PublisherKit.PublisherKitVideoType.PublisherKitVideoTypeScreen);                    
+        } else {
+            Publisher mPublisher = new Publisher.Builder(this.getReactApplicationContext())
+                                        .audioTrack(audioTrack)
+                                        .videoTrack(videoTrack)
+                                        .name(name)
+                                        .audioBitrate(audioBitrate)
+                                        .resolution(Publisher.CameraCaptureResolution.valueOf(resolution))
+                                        .frameRate(Publisher.CameraCaptureFrameRate.valueOf(frameRate))
+                                        .build();
+            if (cameraPosition.equals("back")) {
+                mPublisher.cycleCamera();
+            }
+        }
```

Comment:
Looks good. This can also be written in a concise manner. we can initialize the publisher with the common setting at the beginning. And only set `.capturer(capturer)` or `cameraPosition.equals("back")` inside the `if-else` loop. WDYT?

Reply:
@Lucashuang0802  We can't set the capturer after the Publisher has been initialized.

Reply:
👍